umpost/um2netcdf.py

```diff
@@ -736,6 +727,41 @@ def fix_level_coord(cube, z_rho, z_theta, tol=1e-6):
                 c_sigma.var_name = 'sigma_theta'
 
 
+def fix_pressure_levels(cube, decimals=5):
+    """
+    Reformat pressure level data for NetCDF output.
+
+    This converts units, rounds small fractional errors & ensures pressure is
+    decreasing (following the CMIP6 standard).
+
+    Parameters
+    ----------
+    cube : iris Cube (modifies in place)
+    decimals : number of decimals to round to
+
+    Returns
+    -------
+    None if cube lacks pressure coord or is modified in place, otherwise a new
+    cube if the pressure levels are reversed.
+    """
+    try:
+        pressure = cube.coord('pressure')
+    except iris.exceptions.CoordinateNotFoundError:
+        return
+
+    # update existing cube metadata in place
+    pressure.attributes['positive'] = 'down'
+    pressure.convert_units('Pa')
+
+    # Round small fractions otherwise coordinates are off by 1e-10 in ncdump output
+    pressure.points = np.round(pressure.points, decimals)
+
+    if pressure.points[0] < pressure.points[-1]:
+        # Flip to get pressure decreasing as per CMIP6 standard
+        # NOTE: returns a new cube!
+        return iris.util.reverse(cube, 'pressure')
```

Comment:
Is it possible to ever get a cube where the pressure data isn't monotonic for some reason (including user stupidity)?

Reply:
I'm not sure if the data is always monotonic. 

In the "aiihca.paa1jan" dataset (`gadi:/g/data/tm70/esm15_testing/sample_output/aiihca.paa1jan`), extracting all the pressure coords from 100+ cubes gives 11 copies of these points for `pressure`:

```python
array([   1.,    5.,   10.,   20.,   30.,   50.,   70.,  100.,  150.,
         200.,  250.,  300.,  400.,  500.,  600.,  700.,  850.,  925.,
        1000.])
```
@MartinDix do you know if there is any guarantee that the pressure data is monotonic?

Reply:
Let's throw out an idea for stepwise simplification. The 1st goal for `um2nc-standalone` is to modularise & retrofit unit tests to the code. Changing/extending behaviour is a secondary step, made easier once tests exist. Therefore, an option for this PR/feature is:

* (Optional) add a `TODO: add monotonic test func` reminder to the code
* Create GitHub issue specifically for addressing the monotonic test (do here, add data preprocessing script?)
* Verify this extracted func matches the existing behaviour
* Merge this smaller work unit

How does that sound?

Reply:
Ooops, a `monotonic` check function is included in `iris`:
https://github.com/SciTools/iris/blob/main/lib/iris/util.py#L615

Reply:
Do we know if that check function gets applied to enforce the data to be monotonic, or is it just something we can use in tests?

Reply:
`monotonic()` isn't called directly in `iris.util.reverse()`:
https://github.com/SciTools/iris/blob/e1f5a8b880e443c58cba7770a1ffd32e067a64f2/lib/iris/util.py#L518-L612

Reply:
The test condition here probably needs to be updated to use `monotonic()` then - until we can confirm the pressure values are guaranteed to be monotonic, the current test doesn't actually confirm the direction of the pressure values.

Reply:
> Do we know if that check function gets applied to enforce the data to be monotonic, or is it just something we can use in tests?

As a standalone function, we can use it in both cases.

Reply:
Added TODO to consider monotonic checking in f6ba005c.

Reply:
I've deferred this to #107 to help bring the rest of the work into the `main` branch.